src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -126,6 +128,53 @@ public void saveNewCommentsForGoalById(@PathParam("id") Long goalId, MessageDto
                 .collect(Collectors.toList());
     }
 
+    @POST
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/search-goals")
+    public List<GoalDto> getGoalsByTag(JsonObject searchDto) {
+        List<GoalDto> goalDtoList = new ArrayList<>();
+        for (Map.Entry<String, JsonValue> pair : searchDto.entrySet()) {
+            String adr = pair.getKey();
+            String value = ((JsonString) pair.getValue()).getString();
+            if (adr.equals("goalsearch")) {
+                List<Tag> tagList = goalStore.getTagsByMessage(value);
+                for (Tag tag : tagList) {
+                    List<Goal> goalList = goalStore.getGoalsByTag(tag);
+                    if (goalList.size() != 0) {
+                        goalDtoList = goalList.stream()
+                                .map(DtoConverter::convertGoalToGoalDto)
+                                .collect(Collectors.toList());
+                    } else {
+                        goalDtoList = Collections.emptyList();
+                    }
+                }
+            }
+        }
+        return goalDtoList;
+    }
+
+    @POST
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/search-user")
+    public List<UserSearchDto> getUsersByUsername(JsonObject searchDto) {
+        List<UserSearchDto> userDtoList = new ArrayList<>();
+        for (Map.Entry<String, JsonValue> pair : searchDto.entrySet()) {
+            String adr = pair.getKey();
+            String value = ((JsonString) pair.getValue()).getString();
+            if (adr.equals("usersearch")) {
+                List<User> userList = userStore.getUserByUsername(value);
+                if (userList.size() != 0) {
+                    userDtoList = userList.stream()
+                            .map(DtoConverter::convertUserToUserSearchDto)
+                            .collect(Collectors.toList());
+                } else {
+                    userDtoList = Collections.emptyList();
+                }
+            }
+        }
+        return userDtoList;
+    }
```

Comment:
These two methods look like code duplicates - only the object key is different. Please refactor.